python/ray/workers/default_worker.py

```diff
@@ -1,16 +1,145 @@
 import argparse
 import base64
 import json
+import logging
 import time
 import sys
 import os
+from contextlib import redirect_stdout, redirect_stderr
+from logging.handlers import RotatingFileHandler
 
 import ray
 import ray.actor
 import ray.node
 import ray.ray_constants as ray_constants
 import ray.utils
 from ray.parameter import RayParams
+from ray.ray_logging import StandardStreamInterceptor
+
+
+def setup_and_get_worker_interceptor_logger(is_for_stdout: bool):
+    """Setup a logger to be used to intercept worker log messages.
+
+    NOTE: The method is not idempotent.
+
+    Ray worker logs should be treated in a special way because
+    there's a need to intercept stdout and stderr to support various
+    ray features. For example, ray will prepend 0 or 1 in the beggining
+    of each log message to decide if logs should be streamed to driveres.
+
+    This logger will also setup the RotatingFileHandler for
+    ray workers processes.
+
+    Args:
+        is_for_stdout(bool): True if logger will be used to intercept stdout.
+                             False otherwise.
+    """
+    file_extension = "out" if is_for_stdout else "err"
+    logger = logging.getLogger(f"ray_default_worker_{file_extension}")
```

Comment:
aren't we collapsing these into one file?

Reply:
I just added it to default_worker.py because this method is not really reusable and very specific to default worker (kind of similar to StandardStreamInterceptor, but I thought it was a little bit more general than this method). I am not strongly opponent to move to the ray_logging, so let me know if that's what you want!

Reply:
Sorry, I meant aren't we collapsing the stdout and stderr of ray processes into a single file?

Reply:
Ah, 😂 sorry I was confused for some reason. 

I think that’ll be out of scope of this PR since it hasn’t been internally agreed yet?